src/components/Prescriptions.tsx

```diff
@@ -24,9 +24,20 @@ const Prescriptions: FunctionComponent<PrescriptionsProps> = ({
   const { token } = useContext(TokenContext)
 
   const [filteredPrescriptions, setFilteredPrescriptions] = useState<Prescription[]>(prescriptions)
+  const [searchPatient, setSearchPatient] = useState<string>('')
 
   // Using Record to get a key-value array
-  const [patients, setPatients] = useState<Record<number, Patient | null>>({})
+  const [patients, setPatients] = useState<Record<number, Patient>>({})
+
+  const filterByPrescriptions = useCallback((prescription: Prescription, searchValue: string): boolean => {
+    const doctorMatch = prescription.prescribing_doctor.toLowerCase().includes(searchValue.toLowerCase())
+    const patient = patients[prescription.id]
+    assert(patient)
+    const patientFullName = `${patient.firstname} ${patient.lastname}`.toLowerCase()
+    const patientMatch = patientFullName.includes(searchValue.toLowerCase())
+
+    return doctorMatch || patientMatch
+  }, [patients])
```

Comment:
We could break this down a bit to give it a bit more semantic

Reply:
I created many functions